devtools/cli/src/main/java/io/quarkus/cli/common/RegistryClientMixin.java

```diff
@@ -16,7 +16,7 @@
     final static boolean VALIDATE = !Boolean.parseBoolean(System.getenv("REGISTRY_CLIENT_TEST"));
 
     @CommandLine.Option(names = { "--registry-client" }, description = "Use the Quarkus extension catalog", negatable = true)
-    boolean useRegistryClient = false;
+    boolean useRegistryClient = true;
```

Comment:
will need to add a test (using --dry-run) to make sure the behavior is the same once the default flips. See #18625